src/runtime/actor/envelope.rs

```diff
@@ -0,0 +1,50 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{fmt::Debug, pin::Pin};
+
+use async_trait::async_trait;
+
+use super::{context::ActorContext, Actor};
+
+#[async_trait]
+pub trait HandleEvent<E: Send + Debug>: Actor + Sized {
+    async fn handle_event(
+        &mut self,
+        cx: &mut ActorContext<Self>,
+        event: E,
+        data: &mut Self::Data,
+    ) -> Result<(), Self::Error>;
+}
+
+pub trait DynEvent<A: Actor>: Debug {
```

Comment:
Could you maybe explain in the documentation what's going on here? What is the purpose of `DynEvent`?

Reply:
This is what enables sending any event type to an actor over the same channel as long as they have implemented `HandleEvent` for it.